challenge.rb

```diff
@@ -1,29 +1,67 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  final_result = []
+  intermediate = []
+  input.each do |i|
+  	 intermediate = i.split(//)
+     intermediate[0] = intermediate[0].upcase
+     result = intermediate.join
+   	 final_result.push(result)
+  end
+  final_result
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  result = []
+  input.each_with_index do |animal, index|
+  	 result.push(animal) if index == 2
+  end
+  result
```

Comment:
You want to take an approach that would work even if the dog had a different location in the array.